comp/host-profiler/collector/def/component.go

```diff
@@ -0,0 +1,25 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+//go:build hostprofiler
+
+// Package collector defines the host profiler collector component.
+//
+// The collector component is responsible for collecting host profiling data
+// and sending it to Datadog. It implements the main functionality of the
+// host profiler agent.
+package collector
+
+// team: opentelemetry-agent profiling-full-host
+
+// Component is the component type for the host profiler collector.
+//
+// The collector component provides the main functionality for collecting
+// host profiling data and sending it to Datadog.
+type Component interface {
+	// Run starts the collector and begins collecting host profiling data.
+	// It should be called to start the main collection loop.
+	Run() error
```

Comment:
Why do we need a `Run` method ? (vs using an fx start hook)

Reply:
This is a very good question. 
When using `fxUtil.Run` it is not easier to find the entry point of an application if this entry point uses fx start hook. By having this `Run` function is makes it explicit. See https://github.com/DataDog/datadog-agent/pull/41001/files#diff-30e7d4ca84579cf227c873d5457632440e303e14cb86bceeb08418b9419db4beR45-R49.